scripts/monitoring/auto_manage_ec2_workers.py

```diff
@@ -0,0 +1,172 @@
+import os
+import pytz
+import warnings
+import boto3
+from datetime import datetime
+from dateutil.parser import parse
+from evalai_interface import EvalAI_Interface
+
+warnings.filterwarnings("ignore")
+
+utc = pytz.UTC
+
+aws_keys = {
+    "AWS_ACCOUNT_ID": os.environ.get("AWS_ACCOUNT_ID", "x"),
+    "AWS_ACCESS_KEY_ID": os.environ.get("AWS_ACCESS_KEY_ID", "x"),
+    "AWS_SECRET_ACCESS_KEY": os.environ.get("AWS_SECRET_ACCESS_KEY", "x"),
+    "AWS_REGION": os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+    "AWS_STORAGE_BUCKET_NAME": os.environ.get(
+        "AWS_STORAGE_BUCKET_NAME", "evalai-s3-bucket"
+    ),
+}
+
+
+ENV = os.environ.get("ENV", "dev")
+
+evalai_endpoint = os.environ.get("API_HOST_URL", "http://localhost:8000")
+auth_token = os.environ.get(
+    "AUTH_TOKEN",
+)
+
+
+def get_boto3_client(resource, aws_keys):
+    client = boto3.client(
+        resource,
+        region_name=aws_keys["AWS_REGION"],
+        aws_access_key_id=aws_keys["AWS_ACCESS_KEY_ID"],
+        aws_secret_access_key=aws_keys["AWS_SECRET_ACCESS_KEY"],
+    )
+    return client
+
+
+def get_pending_submission_count(challenge_metrics):
+    pending_submissions = 0
+    for status in ["running", "submitted", "queued", "resuming"]:
+        pending_submissions += challenge_metrics.get(status, 0)
+    return pending_submissions
+
+
+def stop_workers(challenge):
+    target_instance_name = "Worker-Instance-{}-{}".format(
+        ENV, challenge["id"]
+    )
+
+    ec2 = get_boto3_client("ec2", aws_keys)
+    response = ec2.describe_instances(
+        Filters=[
+            {"Name": "tag:Name", "Values": [target_instance_name]},
+            {"Name": "instance-state-name", "Values": ["running"]},
+        ]
+    )
+
+    instances = [
+        instance
+        for reservation in response["Reservations"]
+        for instance in reservation["Instances"]
+    ]
+
+    if instances:
+        for instance in instances:
```

Comment:
I don't think you should hard code the instance name. Instead, you can fetch the instance using `ec2_instance_id` directly and use that. Print the name of the instance you are scaling up or down.

Just check if the instance is in `Running` state or not. If it is `Running`, then stop.